WhateverGreen/kern_igfx.hpp

```diff
@@ -499,6 +499,93 @@ class IGFX {
 		static bool forceWakeWaitAckFallback(uint32_t, uint32_t, uint32_t);
 		static void forceWake(void*, uint8_t set, uint32_t dom, uint32_t);
 	} ForceWakeWorkaround;
+	
+	/**
+	 *  Interface of a submodule to fix Intel graphics drivers
+	 */
+	class PatchSubmodule {
+	public:
+		/**
+		 *  Virtual destructor
+		 */
+		virtual ~PatchSubmodule() = default;
```

Comment:
When are we ever destructing a patch?

Reply:
Previously, `PatchSubmodule` contains only pure virtual functions. As vit9696 suggests, I should make them non-pure and provide default empty implementations. I thought a virtual destructor is required if I want to make a class behave like an interface in C++. Maybe we no longer require the virtual destructor in this case?

Reply:
What I meant is that patch deallocation newer appears to happen in WhateverGreen anyway.